src/VisualStudio/Core/Def/Implementation/CodeLens/RemoteCodeLensReferencesService.cs

```diff
@@ -221,16 +246,25 @@ private static string GetLineTextOrEmpty(TextLineCollection lines, int index)
                 return null;
             }
 
-            var remoteHostClient = await solution.Workspace.Services.GetService<IRemoteHostClientService>().TryGetRemoteHostClientAsync(cancellationToken).ConfigureAwait(false);
-            if (remoteHostClient == null)
+            var client = await RemoteHostClient.TryGetClientAsync(solution.Workspace, cancellationToken).ConfigureAwait(false);
+            if (client != null)
             {
-                // remote host is not running. this can happen if remote host is disabled.
-                return await CodeLensReferencesServiceFactory.Instance.FindReferenceLocationsAsync(solution, documentId, syntaxNode, cancellationToken).ConfigureAwait(false);
+                var result = await client.TryRunRemoteAsync<IEnumerable<ReferenceLocationDescriptor>>(
+                    WellKnownServiceHubServices.CodeAnalysisService,
+                    nameof(IRemoteCodeLensReferencesService.FindReferenceLocationsAsync),
+                    solution,
+                    new object[] { documentId, syntaxNode.Span },
+                    callbackTarget: null,
+                    cancellationToken).ConfigureAwait(false);
+
+                if (result.HasValue)
+                {
+                    return result.Value;
+                }
             }
 
-            return await remoteHostClient.TryRunCodeAnalysisRemoteAsync<IEnumerable<ReferenceLocationDescriptor>>(
-                solution, nameof(IRemoteCodeLensReferencesService.FindReferenceLocationsAsync),
-                new object[] { documentId, syntaxNode.Span }, cancellationToken).ConfigureAwait(false);
+            // remote host is not running. this can happen if remote host is disabled.
```

Comment:
is it allowed to disable the OOP process?  I cna't find any option for that.

Reply:
It's controlled by an internal optional (vsregedit)

Reply:
Yes. I think it's an undocumented regkey.